src/main/scala/com/webtrends/harness/component/akkahttp/AkkaHttpMulti.scala

```diff
@@ -0,0 +1,99 @@
+package com.webtrends.harness.component.akkahttp
+
+import akka.http.scaladsl.model.{HttpMethod, HttpMethods}
+import akka.http.scaladsl.server.Directives.{entity, provide, path => p, _}
+import akka.http.scaladsl.server.{Directive0, Directive1, ImplicitPathMatcherConstruction, PathMatcher}
+import com.webtrends.harness.command.{BaseCommand, CommandBean}
+import com.webtrends.harness.component.akkahttp.methods.AkkaHttpMethod
+
+/**
+  * Use this class to create a command that can handle any number of endpoints with any
+  * number of HTTP methods in a single class
+  */
+trait AkkaHttpMulti extends AkkaHttpBase { this: BaseCommand =>
+  // Map of endpoint names as keys to endpoint info
+  def allPaths: Map[String, Endpoint]
+
+  // Convenient method for when your path will not have any segments
+  def emptyPath(pth: String) = p(pth) & provide(new AkkaHttpPathSegments {})
+
+  // Method that is called for each endpoint object on addition, can override to do special logic
+  def endpointExtraProcessing(end: Endpoint): Unit = {}
+
+  // Method that adds all routes from allPaths
+  override def createRoutes() = {
+    allPaths.foreach { case (pathName, endpoint) =>
+      var segCount = 0
+      // Split the path into segments and map those to their akka http objects
+      val segs = endpoint.url.split("/").filter(_.nonEmpty).toSeq
+      try {
+        // Combine all segments into an akka path
+        val dir = segs.tail.foldLeft(segs.head.asInstanceOf[Any]) { (x, y) =>
+          y match {
+            case s1: String if s1.startsWith("$") =>
+              segCount += 1
+              (x match {
```

Comment:
I apologize for those case statements where I'm checking for each PathMatcher type, it was a huge pain in the buns to get this to work even in that way